classes/interactable/npc/toad/toad.gd

```diff
@@ -41,29 +42,27 @@ const skin_presets = [
 
 @export_enum("Red", "Green", "Blue", "Yellow", "Purple") var spot = 0: set = set_spot
 @export_range(0, 1) var skin = 0: set = set_skin
-var temp_skin
-var temp_spot
-
-
-func _init():
-	temp_skin = skin
-	temp_spot = spot
 
 
 func set_spot(new_spot):
 	for i in range(2):
 		material.set_shader_parameter("color" + str(i), spot_presets[new_spot][i])
 	spot = new_spot
-	temp_spot = new_spot
 
 
 func set_skin(new_skin):
 	for i in range(4):
 		material.set_shader_parameter("color" + str(i + 2), skin_presets[new_skin][i])
 	skin = new_skin
-	temp_skin = new_skin
 
 
 func _ready():
+	set_spot(spot)
+	set_skin(skin)
```

Comment:
These two function calls aren't necessary--the variables' `set` clauses run them just fine.

Reply:
Those are there so Toads initialize with the correct color when instanced into the editor tree. Without it, you'd need to manually switch the spot and skin properties for the palette to load properly.

Reply:
Ah. That makes sense. Would suggest mentioning that in a code comment, then (so future readers know not to refactor it out).